swagger_parser/lib/src/parser/parser.dart

```diff
@@ -420,37 +431,55 @@ class OpenApiParser {
 
         final requestPathResponses = (requestPath
             as Map<String, dynamic>)[_responsesConst] as Map<String, dynamic>;
+        final additionalName = '$key${path}Response'.toPascal;
         final returnType = _version == OAS.v2
-            ? returnTypeV2(requestPathResponses)
-            : returnTypeV3(requestPathResponses);
+            ? returnTypeV2(
+                requestPathResponses,
+                additionalName: additionalName,
+              )
+            : returnTypeV3(
+                requestPathResponses,
+                additionalName: additionalName,
+              );
         final parameters = _version == OAS.v2
             ? parametersV2(requestPath)
             : parametersV3(requestPath);
-        final requestName = _pathMethodName
-            ? (key + path).toCamel
-            : replaceNotEnglishLetter(
-                  requestPath[_operationIdConst]?.toString(),
-                )?.toCamel ??
-                (key + path).toCamel;
-
-        final summary = requestPath[_summaryConst]?.toString();
-        final description = requestPath[_descriptionConst]?.toString();
-        final fullDescription = switch ((summary, description)) {
+
+        final summary = requestPath[_summaryConst]?.toString().trim();
+        var description = requestPath[_descriptionConst]?.toString().trim();
+        description = switch ((summary, description)) {
           (null, null) => null,
-          (_, null) => summary,
-          (null, _) => description,
+          (_, null) || (_, '') => summary,
+          (null, _) || ('', _) => description,
           (_, _) => '$summary\n\n$description',
         };
 
+        final String requestName;
+
+        if (_pathMethodName) {
+          requestName = (key + path).toCamel;
+        } else {
+          final operationIdName =
+              requestPath[_operationIdConst]?.toString().toCamel;
+          final (_, error) = protectName(operationIdName);
+          if (error != null) {
+            description = '$description\n\n$error';
+            requestName = (key + path).toCamel;
+          } else {
+            requestName = operationIdName ?? (key + path).toCamel;
+          }
+        }
+
         final request = UniversalRequest(
           name: requestName,
-          description: fullDescription,
+          description: description,
           requestType: HttpRequestType.fromString(key)!,
           route: path,
           isMultiPart: isMultiPart,
           isFormUrlEncoded: isFormUrlEncoded,
           returnType: returnType,
           parameters: parameters,
+          isDeprecated: requestPath[_deprecatedConst].toString().toBool(),
```

Comment:
requestPath[_deprecatedConst]?.toString().toBool(),

Reply:
isDeprecated cannot take null, if there is no deprecated parameter, we consider the default to be false.
Working with nullable in code is written the same way: 
```dart
nullable: map[_nullableConst].toString().toBool(),
```

Reply:
ok